### PR TITLE
Vue3 migration

### PR DESCRIPTION
**Pull request by Frizi**
This PR introduces Vue 3, without migrating all component away from class syntax yet.

Most of the changes are quite mechanical, except in places where a little stronger types reported errors (mostly around refs).

- a few very simple components were already migrated to `defineComponent` as a proof of concept
- class components got slightly different imports due to changes in vue-class-components dependency
- big diffs on the templates are mostly due to new slot syntax. Slots now must be used as `<template #slot>`, which often required a single extra nesting level in the template tree.
- Soundtrack controller is no longer a real component, but a `use` function (the file path still stays the same, as we have to find a new organization for those anyway). This serves as an introduction to [Composition API](https://composition-api.vuejs.org/#summary)
- vue-property-decorator must be patched a little to remain compatible with vue 3, but it turned out to be a very minimal change

### CODE REVIEW

**Comment by sneakyweasel** [comment]
Seems fine, gj, only concerns is about a performance decrease after updateCell -> QT Simulation.

**Comment by stared** [comment]
Overall, looks nice. 

With tracking, we need to wait with the release (vide https://github.com/MatteoGabriele/vue-gtag/issues/168#issuecomment-684498923), but it seems than only a week more.

File: src/components/AppNotifications.vue [hunk 1]
```diff
@@ -2,23 +2,20 @@
   <div id="notifications">
     <ul v-if="errors.length" class="errors">
       <li v-for="error in errors" :key="error">{{ error }}</li>
-      <li @click="resetErrors"><router-link to="/">click here to get back</router-link></li>
+      <li @click="RESET_ERRORS"><router-link to="/">click here to get back</router-link></li>
```
Comment (stared) [inline src/components/AppNotifications.vue:5]: Any particular reason for `ALL_CAPS`?

File: src/store/storeInterfaces.ts [hunk 2]
```diff
@@ -1,24 +1,87 @@
...
+  ActionContext,
+  MutationTree,
+  ActionHandler,
+} from 'vuex'
 import firebase from '@/config/firebase'
-import { GameStateEnum, ILevel } from '@/engine/interfaces'
 import GameState from '@/engine/GameState'
-import Cell from '@/engine/Cell'
-import Particle from '@/engine/Particle'
+import { S } from 'quantum-tensors/dist/Gates'
```
Comment (stared) [inline src/store/storeInterfaces.ts:15]: Are you sure it is a correct and necessary line?

**Review by stared — APPROVED** [review]
I see a lot of good work there. Impressed you made it work with Vue3!

* Music does not work for me (this would require changes, but happy to wait with it for the next commit).
* Do we want to keep `@Prop` or is it possible to change it for all elements (is it temporary, or intended?)
* Elements in the encyclopedia do not rotate. (It can wait for a new PR, with Board unification).
* SpeachBubble behaves differently (though, to be fair - this element have never worked well, and most likely need to be rewritten from scratch).

Screenshot 2020-09-10 at 20 00 20

**Pull request merged**